### PR TITLE
Feature managed groups read CRUD

### PR DESCRIPTION
**Pull request by calcaide**
Add read CRUD operation to managed groups.
- The breadcrumbs are not working properly, we will take care of those later.
- The form has no action buttons yet, will be added on a future PR.
- Actions dropdown is working but not (yet) the delete function. That will be implemented on the delete PR.

[**Preview Link**](https://boundary-ui-git-feature-managed-group-read-hashicorp.vercel.app/scopes/global/auth-methods/auth-method-id-0/managed-groups/managed-groups-id-0)

Screen Shot 2021-11-09 at 10 30 00 AM

### CODE REVIEW

**Review by randallmorey — CHANGES_REQUESTED** [review]
The managed group tab should only appear when the auth-method has a type of `oidc`.  Password auth methods don't have managed groups.

As an aside, you might also tweak the mocks such that managed groups are _only generated for OIDC auth methods_.  This isn't critical to the mocks, but makes it a bit more realistic.

**Comment by DhariniJeeva** [comment]
more of a design question - thoughts on having a description for `filter` same as the above fields?

**Comment by calcaide** [comment]
> The managed group tab should only appear when the auth-method has a type of `oidc`. Password auth methods don't have managed groups.
> 
> As an aside, you might also tweak the mocks such that managed groups are _only generated for OIDC auth methods_. This isn't critical to the mocks, but makes it a bit more realistic.

Absolutely! nice point. I already applied both changes.
- Just show the tab 'managed groups' when the type of the auth method is `oidc`. 
- Create managed-groups just on the auth method with type `oidc`.
- Bonus track. I added more than just 1 auth-method.

**Comment by calcaide** [comment]
> more of a design question - thoughts on having a description for `filter` same as the above fields?

@DhariniJeeva that is an excellent point. I did add description to `filter` [based on our documentation](https://www.boundaryproject.io/docs/concepts/domain-model/managed-groups#filter).

@gsusmi @randallmorey What do you guys think? should we keep that description on `filter`? Also, should I add description to type?

Screen Shot 2021-11-09 at 11 28 56 AM

File: ui/admin/app/templates/scopes/scope/auth-methods/auth-method/index.hbs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+<h2>{{@model.type}}</h2>
```
Comment (randallmorey) [inline ui/admin/app/templates/scopes/scope/auth-methods/auth-method/index.hbs:1]: Is this for development purposes?
Screenshot 2021-11-10 at 08 36 02

**Review by randallmorey — CHANGES_REQUESTED** [review]

File: ui/admin/app/templates/scopes/scope/auth-methods/auth-method/index.hbs (hunk 1 above)
Comment (calcaide) [reply ui/admin/app/templates/scopes/scope/auth-methods/auth-method/index.hbs]: That was definitely unintentional :sweat_smile:

**Review by gsusmi — APPROVED** [review]
LGTM. Ty!

File: addons/core/translations/resources/en-us.yaml [hunk 2]
```diff
@@ -354,3 +354,6 @@ scope:
 managed-group:
   title: Managed Group
   title_plural: Managed Groups
+  description: A managed group is a resource that represents a collection of accounts.
```
Comment (randallmorey) [inline addons/core/translations/resources/en-us.yaml:357]: "A managed group represents a collection of accounts."  I'd also mention something about the automated/managed nature via a filter.  BE might be able to help here.

**Review by randallmorey — APPROVED** [review]
Provisional approval.  Had one comment in the code.  It would also be nice to get acceptance tests included with this, but they can come later if that's easier.

File: addons/core/translations/resources/en-us.yaml (hunk 2 above)
Comment (calcaide) [reply addons/core/translations/resources/en-us.yaml]: - Change done.
- Also, asked BE some help with enhancing this description.

**Comment by calcaide** [comment]
> Provisional approval. Had one comment in the code. It would also be nice to get acceptance tests included with this, but they can come later if that's easier.

Definitely all the tests are coming on a separate PR just for testing.

**Pull request merged**